### PR TITLE
🌱 Update tests to use same envtest helper, add eventual checks for cache consistency

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
kind/chore

**What this PR does / why we need it**:
This is a refactor of unit tests, which ensures there is no flakes in unit test and moves all tests to a single envtest utility helper across the repo.

This helper already sets up manager instance for the cluster, allowing to write integration tests on our controllers, including testing watch predicates in conjunction.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits into logical changes
- [ ] includes documentation
- [x] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: internal/controllers/import_controller_test.go [hunk 1]
```diff
@@ -57,38 +59,47 @@ var _ = Describe("reconcile CAPI Cluster", func() {
 	)
 
 	BeforeEach(func() {
+		var err error
+
+		ns, err = testEnv.CreateNamespace(ctx, "commonns")
```
Comment (richardcase) [inline internal/controllers/import_controller_test.go:64]: I had to read the name of the namespace twice, i thought there was a typo :wink:

**Review by richardcase — DISMISSED** [review]

**Review by salasberryfin — APPROVED** [review]

**Comment by salasberryfin** [comment]
I'd say we should have this merged before #362 so we can re-use the new test helpers without generating conflicts.

**Comment by Danil-Grigorev** [comment]
@salasberryfin Agree, let me push a round of changes to include komega use across tests, as the combination of Eventually + cl.Get was a pain point for me. Have some changes in another branch I’d like to propagate here + a discovered issue with predicate logic (we can import any cluster on an error in object lookup…) and it should be ready to merge

File: internal/controllers/import_controller_test.go (hunk 1 above)
Comment (Danil-Grigorev) [reply internal/controllers/import_controller_test.go:64]: It is just a generated name for the namespace, and naming is hard…

**Review by salasberryfin — APPROVED** [review]

**Review by richardcase — APPROVED** [review]

**Pull request merged**